FlowCrypt/src/main/java/com/flowcrypt/email/jetpack/viewmodel/MessagesViewModel.kt

```diff
@@ -445,28 +455,49 @@ class MessagesViewModel(application: Application) : AccountViewModel(application
           )
         )
         val messagesBaseInfo = GmailApiHelper.loadMsgsBaseInfo(
-          getApplication(), accountEntity,
-          localFolder, if (totalItemsCount > 0) labelEntity?.nextPageToken else null
+          context = getApplication(),
+          accountEntity = accountEntity,
+          localFolder = localFolder,
+          nextPageToken = if (totalItemsCount > 0) labelEntity?.nextPageToken else null
         )
+
+        val draftIdsMap = when (messagesBaseInfo) {
+          is ListDraftsResponse -> messagesBaseInfo.drafts?.associateBy(
+            { it.message.id },
+            { it.id }) ?: emptyMap()
+          else -> emptyMap()
+        }
+
+        val messages = when (messagesBaseInfo) {
+          is ListMessagesResponse -> messagesBaseInfo.messages ?: emptyList()
+          is ListDraftsResponse -> messagesBaseInfo.drafts?.map { it.message } ?: emptyList()
+          else -> emptyList()
+        }
+
+        val nextPageToken = when (messagesBaseInfo) {
+          is ListMessagesResponse -> messagesBaseInfo.nextPageToken
+          is ListDraftsResponse -> messagesBaseInfo.nextPageToken
```

Comment:
Does Gmail API have a different endpoint to list Drafts vs list Messages?

Reply:
Yup.

### Method: users.drafts.list
https://developers.google.com/gmail/api/reference/rest/v1/users.drafts/list

```json
{
   "drafts":[
      {
         "id":"string",
         "message":{
            "object (Message)"
         }
      }
   ],
   "nextPageToken":"string",
   "resultSizeEstimate":"integer"
}
```

### Method: users.messages.list
https://developers.google.com/gmail/api/reference/rest/v1/users.messages/list

```json
{
   "messages":[
      {
         "object (Message)"
      }
   ],
   "nextPageToken":"string",
   "resultSizeEstimate":"integer"
}
```

Both of these API calls produce different responses with similar data.